tests/test_gdpr_api/helpers.py

```diff
@@ -0,0 +1,124 @@
+import datetime
+from contextlib import contextmanager
+from dataclasses import dataclass
+from typing import Literal
+from unittest.mock import MagicMock, patch
+
+from django.core.handlers.wsgi import WSGIRequest
+from django.utils.timezone import get_default_timezone
+from helusers.settings import api_token_auth_settings
+from jose import jwk, jwt
+from jose.constants import ALGORITHMS
+
+from tests.helpers import ResponseMock
+from users.models import User
+
+
+@contextmanager
+def patch_oidc_config():
+    issuer = api_token_auth_settings.ISSUER
+    side_effects = [
+        ResponseMock(json_data={"issuer": issuer, "jwks_uri": f"{issuer}/jwks"}),
+        ResponseMock(json_data={"keys": [RSA.public_key]}),
+    ]
+
+    with patch("helusers.oidc.requests.get", side_effect=side_effects) as mock:
+        yield mock
+
+
+def get_gdpr_auth_header(user: User, *, scopes: list[str], loa: Literal["substantial", "high", "low"] = "high") -> str:
+    audience = api_token_auth_settings.AUDIENCE
+    issuer = api_token_auth_settings.ISSUER
+    auth_field = api_token_auth_settings.API_AUTHORIZATION_FIELD
+
+    now = datetime.datetime.now(tz=get_default_timezone())
+    expire = now + datetime.timedelta(days=14)
+
+    jwt_data = {
+        "iss": issuer,
+        "aud": audience,
+        "sub": str(user.uuid),
+        "iat": int(now.timestamp()),
+        "exp": int(expire.timestamp()),
+        "loa": loa,
+        auth_field: scopes,
+    }
+    encoded_jwt = jwt.encode(jwt_data, key=RSA.private_key_pem, algorithm=RSA.jose_algorithm)
+    return f"{api_token_auth_settings.AUTH_SCHEME} {encoded_jwt}"
+
+
+# Following adapted from: https://github.com/City-of-Helsinki/helsinki-profile-gdpr-api/blob/main/tests/keys.py
+
+
+@dataclass
+class Key:
+    jose_algorithm: str
+    private_key_pem: str
+    public_key_pem: str
+    public_key: dict
+
+
+def _build_key(private_pem: str, public_pem: str) -> Key:
+    key = Key(
+        jose_algorithm=ALGORITHMS.RS256,
+        private_key_pem=private_pem,
+        public_key_pem=public_pem,
+        public_key=jwk.construct(public_pem, ALGORITHMS.RS256).to_dict(),
+    )
+
+    # Ensure values are strings and not bytes
+    for name in ["n", "e"]:
+        value = key.public_key[name]
+        if isinstance(value, bytes):
+            key.public_key[name] = value.decode("utf-8")
+
+    return key
+
+
+RSA_PRIVATE_KEY = """-----BEGIN PRIVATE KEY-----
```

Comment:
Could rename this to something like `RSA_PRIVATE_KEY_FOR_TESTING` so it is less likely to cause a false positive for security issue (private key committed to repo) when looked at disregarding the surrounding context of unit tests.